cli-helpers/src/commands/analyse-avl-frequency.ts

```diff
@@ -0,0 +1,165 @@
+                objectDate.isSameOrBefore(dateOptions.endThreshold)
+            ) {
+                dateTimes.push(objectDate);
+            }
+        }
+    }
+
+    return dateTimes;
+};
+
+const generateFrequencyResult = (subscriptionId: string, dateTimes: Dayjs[]): FrequencyResult => {
+    const deltas = new Stats();
+
+    for (let i = 0; i < dateTimes.length; i++) {
+        if (i > 0) {
+            const delta = dateTimes[i].diff(dateTimes[i - 1], "milliseconds") / 1000;
+            deltas.push(delta);
+        }
+    }
+
+    return {
+        subscription_id: subscriptionId,
+        mean: deltas.amean().toFixed(1),
+        median: deltas.median().toFixed(1),
+        "50th_percentile": deltas.percentile(50).toFixed(1),
+        "95th_percentile": deltas.percentile(95).toFixed(1),
+        "99th_percentile": deltas.percentile(99).toFixed(1),
+        range: deltas.range().join("-"),
+    };
+};
+
+const generateResultsFile = (dateString: string, frequencyResults: FrequencyResult[]) => {
+    if (frequencyResults.length === 0) {
+        logger.info("No results to generate");
+        return;
+    }
+
+    const csvConfig = mkConfig({
+        quoteStrings: false,
+        useKeysAsHeaders: true,
+    });
+
+    const csv = generateCsv(csvConfig)(frequencyResults);
+    const csvBuffer = new Uint8Array(Buffer.from(asString(csv)));
+    const filename = `avl-frequencies-${dateString}.csv`;
+    writeFileSync(filename, csvBuffer);
+
+    logger.info(`Results written to ${filename}`);
+};
+
+export const analyseAvlFrequency = new Command("analyse-avl-frequency")
+    .addOption(STAGE_OPTION)
+    .option("-d, --date <date>", "Date as YYYY-MM-DD")
+    .action(async (options) => {
+        const { stage, date: dateString } = await withUserPrompts(options, {
+            stage: { type: "list", choices: STAGES },
+            date: { type: "input", default: getDate().format("YYYY-MM-DD") },
+        });
+
+        if (!/^[0-9]{4}-[0-9]{2}-[0-9]{2}$/.test(dateString)) {
+            logger.error("Date must be in format YYYY-MM-DD");
+            return;
+        }
+
+        const date = getDate(dateString);
+        const dateOptions: DateOptions = {
+            dateString,
+            startThreshold: date.startOf("day").add(8, "hours"),
+            endThreshold: date.startOf("day").add(20, "hours"),
+        };
+
+        const bucketName = `integrated-data-avl-raw-siri-vm-${stage}`;
+        const frequencyResults: FrequencyResult[] = [];
+        let subscriptionCount = 0;
+
+        const s3Client = new S3Client({
+            region: "eu-west-2",
+            ...(stage === "local"
+                ? {
+                      endpoint: "http://localhost:4566",
+                      credentials: {
+                          accessKeyId: "DUMMY",
+                          secretAccessKey: "DUMMY",
+                      },
+                  }
+                : {}),
+        });
```

Comment:
In this case I think it might be better is we create an `s3.ts` file similar to what we do in the shared functions folder - that way all the code related to the S3 SDK is in one place.